### PR TITLE
[risk=low][RW-9445] Don't crash when selecting Dataset export from homepage

### PR DESCRIPTION
**Pull request by jmthibault79**
To reproduce:
* create a dataset
* go to homepage
* open the snowman menu for the dataset in the Recently Accessed Items
* choose Export to Notebook

Crashes on Test, fixed in this PR

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: ui/src/app/pages/data/data-set/export-dataset-modal.tsx [hunk 1]
```diff
@@ -61,9 +57,11 @@ const styles = reactStyles({
   },
 });
 
-export const ExportDatasetModal: (props: Props) => JSX.Element = fp.flow(
-  withCurrentWorkspace()
```
Comment (jmthibault79) [inline ui/src/app/pages/data/data-set/export-dataset-modal.tsx:65]: This was the problem: there's no "current workspace" for resources on the homepage.  Now the workspace is passed explicitly, and most other changes in the PR are in support of that.

File: ui/src/app/pages/data/data-set/export-dataset-modal.spec.tsx [hunk 2]
```diff
@@ -58,9 +57,9 @@ describe('ExportDatasetModal', () => {
     registerApiClient(DataSetApi, datasetApiStub);
 
     workspace = workspaceDataStub;
-    currentWorkspaceStore.next(workspace);
 
     testProps = {
+      workspace,
```
Comment (jmthibault79) [inline ui/src/app/pages/data/data-set/export-dataset-modal.spec.tsx:62]: workspace is now passed via props instead of the currentWorkspaceStore

File: ui/src/app/components/render-resource-card.tsx [hunk 3]
```diff
@@ -34,9 +36,7 @@ function renderResourceCard(props: RenderResourceCardProps) {
     [isConceptSet, () => <ConceptSetResourceCard {...props} />],
     [
       isDataSet,
-      () => (
-        <DatasetResourceCard {...props} inactiveBilling={inactiveBilling} />
-      ),
+      () => <DatasetResourceCard {...{ ...props, inactiveBilling }} />,
```
Comment (jmthibault79) [inline ui/src/app/components/render-resource-card.tsx:39]: unrelated opportunistic change

Comment (dmohs) [reply ui/src/app/components/render-resource-card.tsx:39]: Nice.

**Review by dmohs — APPROVED** [review]

**Review by NehaBroad — APPROVED** [review]
